nikola/plugins/misc/scan_posts.py

```diff
@@ -106,3 +106,7 @@ def scan(self):
                         raise err
 
         return timeline
+
+    def supported_extensions(self):
+        """Return a list of supported file extensions, or None if such a list isn't known beforehand."""
+        return list(set([os.path.splitext(x[0])[1] for x in self.site.config['post_pages']]))
```

Comment:
This can be made simpler. Since we dropped 2.6 support, you can use set comprehensions (`{i for i in x}`). And even if we didn’t, using a generator expression inside `set(i for i in x)` would look better.

```py
        return list({os.path.splitext(x[0])[1] for x in self.site.config['post_pages']})
```